factory/ethElrondBridgeComponents.go

```diff
@@ -155,7 +149,7 @@ func NewEthElrondBridgeComponents(args ArgsEthereumToElrondBridge) (*ethElrondBr
 		return nil, err
 	}
 
-	err = components.createDataGetter()
+	err = components.createDataGetter(args)
```

Comment:
perhaps extract `evmCompatibleChain := args.Configs.GeneralConfig.Eth.Chain` here and only pass the chain to these methods, not all args? or maybe keep the chains into components

Reply:
done, kept in components